File: 02-programacion-basica/06-bucles/scala/ejemplos-guiados/BuclesBasicos.scala
object BuclesBasicos:
  def main(args: Array[String]): Unit =
    var suma = 0

    // El rango 1 to 5 incluye ambos extremos.
    for i <- 1 to 5 do
      suma += i

    var cuenta = 3
    while cuenta > 0 do
      println(s"Cuenta: $cuenta")
      cuenta -= 1

    println(s"Suma 1..5 = $suma")

File: 02-programacion-basica/06-bucles/java/ejemplos-guiados/BuclesBasicos.java
public class BuclesBasicos {
    public static void main(String[] args) {
        int suma = 0;

        // for es apropiado cuando conocemos el rango de iteraciones.
        for (int i = 1; i <= 5; i++) {
            suma += i;
        }

        int cuenta = 3;
        // while depende de una condición que debe cambiar durante la ejecución.
        while (cuenta > 0) {
            System.out.println("Cuenta: " + cuenta);
            cuenta--;
        }

        System.out.println("Suma 1..5 = " + suma);
    }
}
